watertap/unit_models/translator_adm1_asm1.py

```diff
@@ -0,0 +1,463 @@
+and used when constructing these,
+**default** - None.
+**Valid values:** {
+see reaction package for documentation.}""",
+        ),
+    )
+
+    def build(self):
+        """
+        Begin building model.
+        Args:
+            None
+        Returns:
+            None
+        """
+        # Call UnitModel.build to setup dynamics
+        super(TranslatorData, self).build()
+
+        self.i_ec = Param(
+            initialize=0.06,
+            units=pyunits.dimensionless,
+            mutable=True,
+            doc="Nitrogen inert content",
+        )
+
+        # Add State Blocks
+        self.properties_in = self.config.inlet_property_package.build_state_block(
+            self.flowsheet().time,
+            doc="Material properties in incoming stream",
+            defined_state=True,
+            has_phase_equilibrium=False,
+            **self.config.inlet_property_package_args
+        )
+
+        self.properties_out = self.config.outlet_property_package.build_state_block(
+            self.flowsheet().time,
+            doc="Material properties in outgoing stream",
+            defined_state=self.config.outlet_state_defined,
+            has_phase_equilibrium=False,
+            **self.config.outlet_property_package_args
+        )
+
+        # Add ports
+        self.add_port(name="inlet", block=self.properties_in, doc="Inlet Port")
+        self.add_port(name="outlet", block=self.properties_out, doc="Outlet Port")
+
+        mw_n = 14.006 * pyunits.kg / pyunits.kmol
+        mw_c = 12.011 * pyunits.kg / pyunits.kmol
+
+        @self.Constraint(
+            self.flowsheet().time,
+            doc="Equality volumetric flow equation",
+        )
+        def eq_flow_vol_rule(blk, t):
+            return blk.properties_out[t].flow_vol == blk.properties_in[t].flow_vol
+
+        @self.Constraint(
+            self.flowsheet().time,
+            doc="Equality temperature equation",
+        )
+        def eq_temperature_rule(blk, t):
+            return blk.properties_out[t].temperature == blk.properties_in[t].temperature
+
+        @self.Constraint(
+            self.flowsheet().time,
+            doc="Equality pressure equation",
+        )
+        def eq_pressure_rule(blk, t):
+            return blk.properties_out[t].pressure == blk.properties_in[t].pressure
+
+        @self.Constraint(
+            self.flowsheet().time,
+            doc="Equality S_I equation",
+        )
+        def eq_SI_conc(blk, t):
+            return (
+                blk.properties_out[t].conc_mass_comp["S_I"]
+                == blk.properties_in[t].conc_mass_comp["S_I"]
+            )
+
+        @self.Constraint(
+            self.flowsheet().time,
+            doc="Equality X_I equation",
+        )
+        def eq_XI_conc(blk, t):
+            return (
+                blk.properties_out[t].conc_mass_comp["X_I"]
+                == blk.properties_in[t].conc_mass_comp["X_I"]
+            )
+
+        self.readily_biodegradable = Set(
```

Comment:
Should we have another set for components with unchanged concentrations (e.g., `X_I` and others where concentration in = concentration out)?

Then, we could eliminate several lines of code and have something like
```
        @self.Constraint(
            self.flowsheet().time,
            self.unchanged_concentrations
            doc="Equality X_I equation",
        )
        def eq_unchanged_concentrations(blk, t, c):
            return (
                blk.properties_out[t].conc_mass_comp[c]
                == blk.properties_in[t].conc_mass_comp[c]
            )
```

It wouldn't change functionality but might make the code more readable.